lib/fs.js

```diff
@@ -453,13 +449,7 @@ Object.defineProperty(fs.exists, internalUtil.promisify.custom, {
 // TODO(joyeecheung): deprecate the never-throw-on-invalid-arguments behavior
 fs.existsSync = function(path) {
   try {
-    path = getPathFromURL(path);
-    validatePath(path);
-    const ctx = { path };
-    binding.stat(pathModule.toNamespacedPath(path), undefined, ctx);
-    if (ctx.errno !== undefined) {
-      return false;
-    }
+    fs.accessSync(path, fs.FS_OK);
```

Comment:
I think it would be good to use a cached `fs.accessSync` version. That way everything would still work, even if that function gets monkey patched.

Reply:
That should be picked up in a separate PR, I think.

Reply:
Since we do not use any reference so far and now introduce one, it might break. So I would rather do that in this PR, but it's not blocking for me.

Reply:
Yeah, I get the argument but monkeypatching fs is something that already a normal pattern. If we're going to fix it for one method, I'd rather fix it for all of them so there's less inconsistency. I'm happy either way tho. I was going to get this landed but will hold off for now.

Reply:
So, is it ok for me to land this?

Reply:
It is OK to land but I personally would say it is better to use a cached version as we otherwise might introduce a potential issue that was not there before.